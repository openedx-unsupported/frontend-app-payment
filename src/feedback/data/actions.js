export const ADD_MESSAGE = 'ADD_MESSAGE';
export const REMOVE_MESSAGE = 'REMOVE_MESSAGE';

export const addMessage = (code, message, data, severity, fieldName = null) => ({
  type: ADD_MESSAGE,
  payload: {
    code,
    message,
    data,
    severity,
    fieldName,
  },
});

export const removeMessage = id => ({
  type: REMOVE_MESSAGE,
  payload: {
    id,
  },
});
