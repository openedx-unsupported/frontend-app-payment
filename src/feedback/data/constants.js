export const DANGER = 'danger';
export const SUCCESS = 'success';
export const INFO = 'info';
export const WARNING = 'warning';
