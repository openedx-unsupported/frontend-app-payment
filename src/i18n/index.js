import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
