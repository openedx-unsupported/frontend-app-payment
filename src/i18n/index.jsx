import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
