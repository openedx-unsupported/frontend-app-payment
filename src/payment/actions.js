import { utils } from '../common';

const { AsyncActionType } = utils;

export const FETCH_BASKET = new AsyncActionType('PAYMENT', 'FETCH_BASKET');


export const fetchBasket = () => ({
  type: FETCH_BASKET.BASE,
  payload: {},
});

export const fetchBasketBegin = () => ({
  type: FETCH_BASKET.BEGIN,
});

export const fetchBasketSuccess = result => ({
  type: FETCH_BASKET.SUCCESS,
  payload: result,
});

export const fetchBasketFailure = () => ({
  type: FETCH_BASKET.FAILURE,
});
