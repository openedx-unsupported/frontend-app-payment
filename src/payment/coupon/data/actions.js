import { AsyncActionType } from '../../../common/utils';

export const ADD_COUPON = new AsyncActionType('PAYMENT', 'ADD_COUPON');
export const REMOVE_COUPON = new AsyncActionType('PAYMENT', 'REMOVE_COUPON');
export const UPDATE_COUPON_DRAFT = 'PAYMENT__UPDATE_COUPON_DRAFT';

// ADD COUPON

export const addCoupon = code => ({
  type: ADD_COUPON.BASE,
  payload: {
    code,
  },
});

export const addCouponBegin = () => ({
  type: ADD_COUPON.BEGIN,
});

export const addCouponSuccess = (voucherId, code, benefit) => ({
  type: ADD_COUPON.SUCCESS,
  payload: {
    benefit,
    code,
    voucherId,
  },
});

export const addCouponFailure = errorCode => ({
  type: ADD_COUPON.FAILURE,
  payload: {
    errorCode,
  },
});

// REMOVE COUPON

export const removeCoupon = voucherId => ({
  type: REMOVE_COUPON.BASE,
  payload: {
    voucherId,
  },
});

export const removeCouponBegin = () => ({
  type: REMOVE_COUPON.BEGIN,
});

export const removeCouponSuccess = result => ({
  type: REMOVE_COUPON.SUCCESS,
  payload: {
    result,
  },
});

export const removeCouponFailure = errorCode => ({
  type: REMOVE_COUPON.FAILURE,
  payload: {
    errorCode,
  },
});

// UPDATE COUPON DRAFT

export const updateCouponDraft = code => ({
  type: UPDATE_COUPON_DRAFT,
  payload: {
    code,
  },
});
