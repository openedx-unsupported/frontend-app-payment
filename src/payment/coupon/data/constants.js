export const PERCENTAGE_BENEFIT = 'Percentage';
export const ABSOLUTE_BENEFIT = 'Absolute';
