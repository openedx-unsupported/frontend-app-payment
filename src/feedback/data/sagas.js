import { put } from 'redux-saga/effects';

import { addMessage, clearMessages } from './actions';
import { MESSAGE_TYPES } from './constants';

export function* handleErrors(e, clearExistingMessages) {
  if (clearExistingMessages) {
    yield put(clearMessages());
  }

  // If this doesn't contain anything we understand, add a fallback error message
  if (e.errors === undefined && e.fieldErrors === undefined && e.messages === undefined) {
    yield put(addMessage('fallback-error', null, {}, MESSAGE_TYPES.ERROR));
  }
  if (e.errors !== undefined) {
    for (let i = 0; i < e.errors.length; i++) { // eslint-disable-line no-plusplus
      const error = e.errors[i];
      if (error.code === 'sku-error-message') {
        yield put(addMessage(error.code, error.userMessage, {}, MESSAGE_TYPES.ERROR));
      } else if (error.data === undefined && error.messageType === null) {
        yield put(addMessage('transaction-declined-message', error.userMessage, {}, MESSAGE_TYPES.ERROR));
      } else {
        yield put(addMessage(error.code, error.userMessage, error.data, error.messageType));
      }
    }
  }
  if (e.messages !== undefined) {
    for (let i = 0; i < e.messages.length; i++) { // eslint-disable-line no-plusplus
      const message = e.messages[i];
      yield put(addMessage(message.code, message.userMessage, message.data, message.messageType));
    }
  }
}

export function* handleMessages(messages, clearExistingMessages, url) {
  // If this doesn't contain anything we understand, bail.
  if (!Array.isArray(messages)) {
    return null;
  }

  if (clearExistingMessages) {
    yield put(clearMessages());
  }

  // Display an error message if one is passed through as a url parameter
  if (url && url.indexOf('error_message') > 0) {
    const errorMessage = new URLSearchParams(url).get('error_message');
    yield put(addMessage('error_message', `${errorMessage}`, {}, MESSAGE_TYPES.ERROR));
  }

  for (let i = 0; i < messages.length; i++) { // eslint-disable-line no-plusplus
    const message = messages[i];
    yield put(addMessage(message.code, message.userMessage, message.data, message.messageType));
  }
  return null;
}
