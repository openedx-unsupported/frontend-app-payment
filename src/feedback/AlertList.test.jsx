import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer, { act } from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import ConnectedAlertList from './AlertList';
import createRootReducer from '../data/reducers';
import { addMessage } from './data/actions';
import { MESSAGE_TYPES } from './data/constants';

jest.mock('@edx/frontend-platform/logging', () => ({
  logError: jest.fn(),
}));

describe('AlertList', () => {
  let store;

  beforeEach(() => {
    store = createStore(createRootReducer());
  });

  it('should be null by default', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider store={store}>
          <ConnectedAlertList removeMessage={jest.fn()} />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render messages of each type', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider store={store}>
          <ConnectedAlertList
            messageCodes={{
              boo: 'Boo indeed!',
              gah_error: 'Gah!  An error!',
            }}
            removeMessage={jest.fn()}
          />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component);
    act(() => {
      store.dispatch(addMessage('boo', null, { needed: 'data' }, MESSAGE_TYPES.WARNING));
      store.dispatch(addMessage('bah', 'Bah!', null, MESSAGE_TYPES.INFO));
      store.dispatch(addMessage(null, 'Meh!', { needed: 'data' }, MESSAGE_TYPES.SUCCESS));
      store.dispatch(addMessage('gah_error', null, { needed: 'data' }, MESSAGE_TYPES.ERROR, 'gah'));
      store.dispatch(addMessage(null, 'Debug debug', null, MESSAGE_TYPES.DEBUG));
      store.dispatch(addMessage('fallback-error', null, null, MESSAGE_TYPES.ERROR));
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
