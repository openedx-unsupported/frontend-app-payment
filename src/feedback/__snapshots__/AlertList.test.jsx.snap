// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertList should be null by default 1`] = `null`;

exports[`AlertList should render messages of each type 1`] = `
Array [
  <div
    className="fade alert-content alert alert-warning show"
    dialog="Boo indeed!"
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        Boo indeed!
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
  <div
    className="fade alert-content alert alert-info show"
    dialog="Bah!"
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        Bah!
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
  <div
    className="fade alert-content alert alert-success show"
    dialog="Meh!"
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        Meh!
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
  <div
    className="fade alert-content alert alert-danger show"
    dialog="Gah!  An error!"
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        Gah!  An error!
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
  <div
    className="fade alert-content alert alert-warning show"
    dialog="Debug debug"
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        Debug debug
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
  <div
    className="fade alert-content alert alert-danger show"
    dialog={
      <FallbackErrorMessage
        values={Object {}}
      />
    }
    role="alert"
  >
    <div
      className="pgn__alert-message-wrapper"
    >
      <div
        className="alert-message-content"
      >
        There was an unexpected problem. If the problem persists, please 
        <a
          className="pgn__hyperlink default-link standalone-link"
          href="http://localhost:18000/support"
          onClick={[Function]}
          target="_self"
        >
          contact support
        </a>
        .
      </div>
      <div
        className="pgn__alert-actions pgn__action-row"
      >
        <span
          className="pgn__action-row-spacer"
        />
        <button
          className="btn btn-tertiary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Dismiss
        </button>
      </div>
    </div>
  </div>,
]
`;
